flit_core/flit_core/config.py

```diff
@@ -510,6 +510,12 @@ def _check_list_of_str(d, field_name):
             "{} field should be a list of strings".format(field_name)
         )
 
+def normalize_pkg_name(name:str)->str:
+    if name.endswith('-stubs'):
+        return name.removesuffix('-stubs').replace('-','_') + '-stubs'
```

Comment:
The special casing should be explained here. For example, typeshed uses a `types-` prefix, which this doesn't cover. Is that important?

Reply:
it's from pep-0561, a type stub package should add `-stubs` to main package.

the `types-` prefix is just the pypi name, which is not the director contains pyi files.

For example, if you unzip the `types_six-1.17.0.20250403-py3-none-any.whl`, you will get this:

```text
Archive:  types_six-1.17.0.20250403-py3-none-any.whl
  inflating: six-stubs/METADATA.toml
  inflating: six-stubs/__init__.pyi
  inflating: six-stubs/py.typed
  inflating: six-stubs/moves/BaseHTTPServer.pyi
  inflating: six-stubs/moves/CGIHTTPServer.pyi
  inflating: six-stubs/moves/SimpleHTTPServer.pyi
  inflating: six-stubs/moves/__init__.pyi
  inflating: six-stubs/moves/_dummy_thread.pyi
...
```

Reply:
It's still possible that a package named as something like `pkg-stubs2` and contins `pkg-stubs/__init__.pyi` for package "pkg", but I don't think it's worth handling ...

Reply:
Just tested this branch and `types-pkg` would also works as a package name (if `tool.flit.module` is used). The only important thing for PEP 561 as I recall is that the package dir name is `{original_name}-stubs`. E.g. `foopkg-stubs` for `foopkg`.

Reply:
Actually any package name works, if `tool.flit.module` is given. this is how I build `foopkg-stubs`  with flit-core currently.